hasher.go

```diff
@@ -144,8 +144,8 @@ func (n *Hasher) EmptyRoot() []byte {
 }
 
 // HashLeaf hashes leaves to:
-// ns(rawData) || ns(rawData) || hash(leafPrefix || rawData), where raw data is the leaf's
-// data minus the namespaceID (namely leaf[NamespaceLen:]).
+// ns(leaf) || ns(leaf) || hash(leafPrefix || leaf), where ns(leaf) is the namespaceID
+// inside the leaf's data namely leaf[:n.NamespaceLen]).
```

Comment:
Thanks, this is clearer as it is aligned with the naming of the actual variable (`leaf`). I think the original comment was a somewhat confusing attempt to say that the `leaf` contains both the namespace + the raw data. This is still implicitly there ("where ns(leaf) is the namespaceID inside the leaf's data namely leaf[:n.NamespaceLen])".